apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -84,9 +83,6 @@ type ProvisionerAPI struct {
 	// Hold on to the controller UUID, as we'll reuse it for a lot of
 	// calls.
 	controllerUUID string
-
-	// Used for MaybeWriteLXDProfile()
-	mu sync.Mutex
```

Comment:
This actually didn't do what we thought it did.

If there was two connections to the provisioner, they could write the LXD profile independently then the mutex wouldn't be global. This was probably a reason why we got bugs sometimes about writing LXD profiles.